dynamic_environment/resource/hook/dynamic_environment.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import json
+import os
+import re
+import ftrack
+import ftrack_api
+
+logging.basicConfig()
+logger = logging.getLogger()
+
+
+def load_env(path):
+    """Load options json from path"""
+
+    with open(path, "r") as f:
+        return json.load(f)
+
+
+def appendPath(path, key, environment):
+    '''Append *path* to *key* in *environment*.'''
+    try:
+        environment[key] = (
+            os.pathsep.join([
+                environment[key], str(path)
+            ])
+        )
+    except KeyError:
+        environment[key] = str(path)
+
+    return environment
+
+
+def dynamic_environment(event):
+    '''Modify the application environment.'''
+    data = event['data']
+    app = data['application']
+
+    logger.info('APP INFO:{}'.format(app))
+
+    taskid = data['context']['selection'][0].get('entityId')
+    session = ftrack_api.Session()
+    task = session.query('Task where id is {0}'.format(taskid)).one()
+
+    # get location of json environmnets from env var
+    env_path = os.environ.get('FTRACK_APP_ENVIRONMENTS')
+    env_files = []
+
+    # determine config file for version independent environment
+    app_name, app_variant = app['identifier'].split('_')
+    app_file_name = '{}.json'.format(app_name)
+    env_files.append(os.path.join(env_path, app_file_name))
```

Comment:
Got a ```TypeError: object of type 'NoneType' has no len()``` because I didn't set the environment variable ```FTRACK_APP_ENVIRONMENT``` I'm guessing. Maybe throw a better error if the environment variable is not set?